internal/pkg/mount/manager/virtual/virtual.go

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package virtual
+
+import (
+	"github.com/talos-systems/talos/internal/pkg/mount"
+	"golang.org/x/sys/unix"
+)
+
+// MountPoints returns the mountpoints required to boot the system.
+func MountPoints() (mountpoints *mount.Points, err error) {
+	virtual := mount.NewMountPoints()
+	virtual.Set("dev", mount.NewMountPoint("devtmpfs", "/dev", "devtmpfs", unix.MS_NOSUID, "mode=0755"))
+	virtual.Set("proc", mount.NewMountPoint("proc", "/proc", "proc", unix.MS_NOSUID|unix.MS_NOEXEC|unix.MS_NODEV, ""))
+	virtual.Set("sys", mount.NewMountPoint("sysfs", "/sys", "sysfs", 0, ""))
+	virtual.Set("run", mount.NewMountPoint("tmpfs", "/run", "tmpfs", 0, ""))
+	virtual.Set("tmp", mount.NewMountPoint("tmpfs", "/tmp", "tmpfs", 0, ""))
```

Comment:
not for this pr.
curious if we should set a limit for these tmpfs filesystems. since they're entirely backed by memory it makes me al little concerned a rogue process could 💣 💥 a system

Reply:
We can certainly set the size. Agree that is for a followup as I'm not sure how big we should go just yet, or if we should base it on percentage.